validator/src/main.rs

```diff
@@ -1396,6 +1396,22 @@ pub fn main() {
         } else {
             vec![ledger_path.join("accounts")]
         };
+
+    let account_paths: Vec<PathBuf> = account_paths
+        .into_iter()
+        .map(|account_path| {
+            match std::fs::create_dir_all(&account_path)
```

Comment:
Creation happens before canonicalization; this is intentional.

std::fs::canonicalize will fail if the path does not exist. So we have 4 cases:
1. user passes a path which does not exist -> we create it, canonicalization just makes it absolute.
2. user passes a path which does exist -> "creating" does nothing, canonicalization just makes it absolute.
3. user passes a broken symlink -> canonicalization fails, validator crashes
4. user passes a good symlink -> create does nothing, canonicalization gives us the followed link

Reply:
To be clear I just re-implemented the logic which was removed in the previous PR

Reply:
should have commented after I create the function. Same comments applied to the `create_and_canonicalize_directories` fn